lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -2611,107 +2621,242 @@ bool AdjointGen::run() {
 
 namespace {
 class AdjointEmitter;
+class AdjointValue;
 
-/// A symbolic adjoint value that is capable of representing zero value 0 and
-/// 1, in addition to a materialized SILValue. This is expected to be passed
-/// around by value in most cases, as it's two words long.
-class AdjointValue {
-  friend class AdjointEmitter;
+class Cleanup {
```

Comment:
Could you add a doc comment for `Cleanup` and `ValueWithCleanup` explaining their purpose/usage?